cypress/integration/core-opensearch-dashboards/opensearch-dashboards/workspace-plugin/mds_workspace_acl.spec.js

```diff
@@ -0,0 +1,239 @@
+    ...getPolicy('library_read', NONE_DASHBOARDS_ADMIN_USERNAME),
+  },
+  [libraryWriteWorkspaceName]: {
+    ...getPolicy('read', NONE_DASHBOARDS_ADMIN_USERNAME),
+    ...getPolicy('library_write', NONE_DASHBOARDS_ADMIN_USERNAME),
+  },
+  [ownerWorkspaceName]: {
+    ...getPolicy('write', NONE_DASHBOARDS_ADMIN_USERNAME),
+    ...getPolicy('library_write', NONE_DASHBOARDS_ADMIN_USERNAME),
+  },
+};
+
+const setupWorkspace = (workspaceName, datasourceId) => {
+  return cy
+    .createWorkspace({
+      name: workspaceName,
+      settings: {
+        ...(datasourceId ? { dataSources: [datasourceId] } : {}),
+        permissions: ACLPolicyMap[workspaceName],
+      },
+    })
+    .then((value) => {
+      // load sample data
+      cy.loadSampleDataForWorkspace('ecommerce', value, datasourceId);
+      cy.wrap(value);
+    });
+};
+
+if (
+  Cypress.env('WORKSPACE_ENABLED') &&
+  Cypress.env('SAVED_OBJECTS_PERMISSION_ENABLED') &&
+  Cypress.env('SECURITY_ENABLED')
+) {
+  describe('Workspace ACL', () => {
+    const originalUser = ADMIN_AUTH.username;
+    const originalPassword = ADMIN_AUTH.password;
+    before(() => {
+      cy.deleteWorkspaceByName(noPermissionWorkspaceName);
+      cy.deleteWorkspaceByName(readOnlyWorkspaceName);
+      cy.deleteWorkspaceByName(libraryWriteWorkspaceName);
+      cy.deleteWorkspaceByName(ownerWorkspaceName);
+
+      cy.createInternalUser(NONE_DASHBOARDS_ADMIN_USERNAME, workspaceTestUser);
+      cy.createRole(WORKSPACE_TEST_ROLE_NAME, workspaceTestRole);
+      cy.createRoleMapping(WORKSPACE_TEST_ROLE_NAME, {
+        users: [NONE_DASHBOARDS_ADMIN_USERNAME],
+      });
+
+      if (Cypress.env('DATASOURCE_MANAGEMENT_ENABLED')) {
+        cy.createDataSourceNoAuth().then((result) => {
+          datasourceId = result[0];
+          expect(datasourceId).to.be.a('string').that.is.not.empty;
+          setupWorkspace(noPermissionWorkspaceName, datasourceId).then(
+            (value) => (noPermissionWorkspaceId = value)
+          );
+          setupWorkspace(readOnlyWorkspaceName, datasourceId).then(
+            (value) => (readOnlyWorkspaceId = value)
+          );
+          setupWorkspace(libraryWriteWorkspaceName, datasourceId).then(
+            (value) => (libraryWriteWorkspaceId = value)
+          );
+          setupWorkspace(ownerWorkspaceName, datasourceId).then(
+            (value) => (ownerWorkspaceId = value)
+          );
+        });
+      } else {
+        setupWorkspace(noPermissionWorkspaceName, datasourceId).then(
+          (value) => (noPermissionWorkspaceId = value)
+        );
+        setupWorkspace(readOnlyWorkspaceName, datasourceId).then(
+          (value) => (readOnlyWorkspaceId = value)
+        );
+        setupWorkspace(libraryWriteWorkspaceName, datasourceId).then(
+          (value) => (libraryWriteWorkspaceId = value)
+        );
+        setupWorkspace(ownerWorkspaceName, datasourceId).then(
+          (value) => (ownerWorkspaceId = value)
+        );
```

Comment:
Nit: Should them be put together into a function?

Reply:
Yeah, done for that.